web3/exceptions.py

```diff
@@ -197,6 +197,15 @@ class InvalidEventABI(ValueError):
 class SolidityError(ValueError):
     # Inherits from ValueError for backwards compatibility
     """
-    Raised on a solidity require/revert
+    Raised on a contract revert error
+    """
+    pass
+
+
+class RevertTransaction(SolidityError, ValueError):
```

Comment:
I wonder if something more like `ContractError` or `ContractLogicException` would be more descriptive

Reply:
We've done an admirable job of bike shedding on this one, but I do find it appealing to lump all asserts, reverts, requires, and related future keywords into one exception type.

Reply:
Yeah, it makes sense to me because it's sort of like "this is an intended application exception" vs. "this is an unexpected VM exception"

Reply:
I'm good with that. Will change in a bit!